frontend/src/lib/components/PropertiesTable/PropertiesTable.tsx

```diff
@@ -85,17 +86,23 @@ function ValueDisplay({
     const valueComponent = (
         <span
             className={clsx(
-                'relative inline-flex items-center flex flex-row flex-nowrap w-fit break-all',
+                'relative inline-flex gap-1 items-center flex flex-row flex-nowrap w-fit break-all',
                 canEdit ? 'editable ph-no-capture' : 'ph-no-capture'
             )}
             onClick={() => canEdit && textBasedTypes.includes(valueType) && setEditing(true)}
         >
             {!isURL(value) ? (
-                valueString
+                <>
+                    <span>{valueString}</span>
+                    {canEdit && <IconPencil />}
+                </>
             ) : (
-                <Link to={value} target="_blank" className="value-link" targetBlankIcon>
-                    {valueString}
-                </Link>
+                <>
+                    <Link to={value} target="_blank" className="value-link" targetBlankIcon>
+                        {valueString}
+                    </Link>
+                    {canEdit && <IconPencil />}
+                </>
             )}
```

Comment:
Or better yet - remove the icon from both conditionals and add it afterwards

```suggestion
            )}
            {canEdit && <IconPencil />}
```